src/backend/columnar/columnar_reader.c

```diff
@@ -1010,12 +1010,11 @@ static bool *
 ProjectedColumnMask(uint32 columnCount, List *projectedColumnList)
 {
 	bool *projectedColumnMask = palloc0(columnCount * sizeof(bool));
-	ListCell *columnCell = NULL;
+	ListCell *lc;
 
-	foreach(columnCell, projectedColumnList)
+	foreach(lc, projectedColumnList)
 	{
-		Var *column = (Var *) lfirst(columnCell);
-		uint32 columnIndex = column->varattno - 1;
+		uint32 columnIndex = lfirst_int(lc) - 1;
```

Comment:
I think it would be nice to clarify why we do -1 here (as in my other suggestions)

Reply:
Done.